### PR TITLE
TextEntry Patches (1.5.x)

### PR DESCRIPTION
**Pull request by onuelito**
- added `_active_entry` property to `Frame` class for the purpose of updating `TextEntry` regardless of the mouse being far from it
 - removed and added lines to `TextEntry` class for the same purpose as above

### CODE REVIEW

File: pyglet/gui/frame.py [hunk 1]
```diff
@@ -87,49 +88,84 @@ def remove_widget(self, widget):
...
         """Pass the event to any widgets within range of the mouse"""
+
+        if self._active_entry: 
+            self._active_entry.on_mouse_press(x, y, buttons, modifiers)
+            self._active_entry = self._active_entry if self._active_entry._focus else None
+        
+
         for widget in self._cells.get(self._hash(x, y), set()):
-            widget.on_mouse_press(x, y, buttons, modifiers)
+            if type(widget).__name__ == 'TextEntry':
```
Comment (Lattay) [inline pyglet/gui/frame.py:98]: This should either be `if type(widget) == TextEntry` or `if isinstance(widget, TextEntry)` (more likely the latter).

File: pyglet/gui/frame.py [hunk 2]
```diff
@@ -87,49 +88,84 @@ def remove_widget(self, widget):
...
         for widget in self._active_widgets:
             widget.on_mouse_release(x, y, buttons, modifiers)
         self._active_widgets.clear()
 
     def on_mouse_drag(self, x, y, dx, dy, buttons, modifiers):
         """Pass the event to any widgets that are currently active"""
+        if self._active_entry: self._active_entry.on_mouse_drag(x, y, dx, dy, buttons, modifiers)
+
         for widget in self._active_widgets:
+            if widget == self._active_entry: continue
```
Comment (Lattay) [inline pyglet/gui/frame.py:120]: Can you explain this special case in `mouse_drag` but not in `mouse_release`?

File: pyglet/gui/frame.py [hunk 3]
```diff
@@ -87,49 +88,84 @@ def remove_widget(self, widget):
...
+
+        if self._active_entry: 
+            self._active_entry.on_mouse_press(x, y, buttons, modifiers)
+            self._active_entry = self._active_entry if self._active_entry._focus else None
+        
+
         for widget in self._cells.get(self._hash(x, y), set()):
-            widget.on_mouse_press(x, y, buttons, modifiers)
+            if type(widget).__name__ == 'TextEntry':
+                if self._active_entry == widget: continue
```
Comment (Lattay) [inline pyglet/gui/frame.py:99]: Single line if blocks are very unusual in Python, and they don't seem to be used anywhere in pyglet. Maybe use black for automatic formatting.

**Review by Lattay — COMMENTED** [review]
Hi,
I am not (yet?) a contributor of Pyglet, but I am also interested in the GUI module and planning on contributing to it, so I read your commit. I have a few suggestions that may improve your contribution.

File: pyglet/gui/frame.py (hunk 2 above)
Comment (onuelito) [reply pyglet/gui/frame.py:120]: I forgot to add the line

**Pull request closed without merging**